spec/requests/openid_google_provider_callback_spec.rb

```diff
@@ -0,0 +1,87 @@
+                  "prompt" => "none" }
+    uri = URI('/auth/google/callback')
+    uri.query = URI.encode_www_form([['code', auth_hash['code']],
+                                     ['state', auth_hash['state']],
+                                     ['scope', auth_hash['scope']],
+                                     ['authuser', auth_hash['authuser']],
+                                     ['prompt', auth_hash['prompt']]])
+    stub_request(:post, "https://accounts.google.com/o/oauth2/token").to_return(
+      status: 200,
+      body: {
+        "access_token" =>
+        "ya29.a0Ael9sCPGoZQiKuMHHVKiaiWV9NatII8T7ZY6XiwTcY-VtvSnmPH53BXDoWGU7OpFY7ctZjY0Qf-Cd_5HHULGoF_m-3WEgMvuO7F11nbYI7qoe95enqneFgDh__vvTxGRAGPpl_Xf7qbXVznh35-DHuvhyPAZmMwaCgYKAQISARASFQF4udJhMeehVtS01I8wd8HL6ReQDw0166",
+        "expires_in" => 3594,
+        "scope" =>
+        "https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/userinfo.profile openid",
+        "token_type" => "Bearer",
+        "id_token" =>
+        "eyJhbGciOiJSUzI1NiIsImtpZCI6IjFhYWU4ZDdjOTIwNThiNWVlYTQ1Njg5NWJmODkwODQ1NzFlMzA2ZjMiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiNDI3NzUwNzQ4MTg2LWQ4OGozamNlYmN2bGlxMmd0a3RiZm1oc2lhNjYxZDU4LmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiNDI3NzUwNzQ4MTg2LWQ4OGozamNlYmN2bGlxMmd0a3RiZm1oc2lhNjYxZDU4LmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTA3NDAzNTExMDM3OTIxMzU1MzA3IiwiZW1haWwiOiJiYTFhc2hwYXNoQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhdF9oYXNoIjoiVFBtc0ZHRng4cjdrb3RiZkJud0xVdyIsImlhdCI6MTY4MDYxMjE5NCwiZXhwIjoxNjgwNjE1Nzk0fQ.IDKlHDVg1d7tAqb8eRiq90T52xnwVX9huDjpdLoJpqr4xlnTrFCdalxJBBHd9Cv39g2KPuJaCU21B59yNAyJP6bl5P8e9Ky-y8wOFcgHqcG5qXcNtxCS3imASCchRTtre8yp9AQGYkTIC0Jh6lWg0trdfO-_idKBsd5naJeaeYdeZGkpQ8D4dxn_odla67BO3y2mUtyE4gEbzyq6wTXDATN4ucM4Dyp3Wdk7YpYYuFN1g-sF6NFl4YqugQ4zk-pYYtPLlPgGiqi3_hO9kYbRDhNBtfbMx568m-CyM2tiOIkb4utPR20scSiRqnY2oxOcd5g9znvJOjtanHM3KVdj5g"
+      }.to_json,
+      headers: { "content-type" => "application/json; charset=utf-8" }
+    )
+    stub_request(:get, "https://www.googleapis.com/oauth2/v3/userinfo?alt=json").to_return(
+      status: 200,
+      body: { "sub" => "107403511037921355307",
+              "name" => "Firstname Lastname",
+              "given_name" => "Firstname",
+              "family_name" => "Lastname",
+              "picture" => "https://lh3.googleusercontent.com/a/AGNmyxZtDAl-mgOOCF_DCo-WWEct-LyVp7zGhXkfKR8r=s96-c",
+              "email" => "email@dummy.com",
+              "email_verified" => true,
+              "locale" => "en-GB" }.to_json,
+      headers: { "content-type" => "application/json; charset=utf-8" }
+    )
+    allow_any_instance_of(OmniAuth::Strategies::OpenIDConnect).to receive(:session) {
+                                                                    { 'omniauth.state' => auth_hash['state'] }
+                                                                  }
+
+    response = get uri.to_s
```

Comment:
Everything inside the it block up until this line (exclusive) should go into a before block. Also the hashes for the stubs should be defined in let statements.

Reply:
Usually when there is only one test case I postpone the extraction into `let`s and `before` block and use local variable and scope.
Sure, I will do it.